daliuge-translator/dlg/dropmake/lg_node.py

```diff
@@ -232,15 +257,19 @@ def gid(self):
         else:
             return self.group.id
 
-    def add_output(self, lg_node):
+    def add_output(self, lg_node, srcPort=None):
         if lg_node not in self._outputs:
             self._outputs.append(lg_node)
+        if self.jd.get("outputPorts") and srcPort is not None and srcPort in self.jd["outputPorts"]:
+            self.jd["outputPorts"][srcPort]["target_id"] = lg_node.id
 
-    def add_input(self, lg_node):
+    def add_input(self, lg_node, tgtPort=None):
         # only add if not already there
         # this may happen in nested constructs
         if lg_node not in self._inputs:
             self._inputs.append(lg_node)
+        if self.jd.get("inputPorts") and tgtPort is not None and tgtPort in self.jd["inputPorts"]:
```

Comment:
Is there a reason we are storing this information in two places now? The `self._inputs/self._outputs` data structure already exists, and what this new code is doing is a) duplicating the data into a second data structure, and b) adding further coupling with the `jd` dictionary. 

If we want things stored as dicts, we should modify `self._inputs`, not create a _two_ separate places where we can access the data (which is an issue we already have across the code). 

If it is too much of a change to go through and convert `self._inputs` to a dictionary, I would advocate for using the self._inputs code, add a helper function `get_input_id` which grabs the id from the list, and then we should raise a ticket to address converting the data structure to a dictionary.